lib/mm_network.pm

```diff
@@ -66,11 +65,9 @@ sub configure_dhcp {
     type_string "NIC=`grep \$MAC /sys/class/net/*/address |cut -d / -f 5`;";
     type_string("echo \"STARTMODE='auto'\nBOOTPROTO='dhcp'\n\" > /etc/sysconfig/network/ifcfg-\$NIC;");
     type_string("done\n");
```

Comment:
> As long as there is only command execution by typing in a shell typing faster than a prompt reappearance should not harm. Some characters might appear in front of the prompt sign but as soon as the prompt shows up the content will be echoed after it correctly and also correctly executed.

From writing the LTP test runner, I am 90% certain this is not true. It might even vary depending on how the TTY/PTY driver and line disciplines handle file descriptors. Some shells might deliberately drop input. So I would just assume that we have to wait.

> Only if an application catching stdin would be called in the meantime this would be a problem

I have no idea why, but it appears `echo` captures stdin, or at least, stdin disappears into a black hole at some point during a call to `echo` (this will vary across shells because `echo` is sometimes a shell builtin function). In this particular case, type_string("rcnetwork restart\n") will probably drop characters due to the previous call to `echo`.

> I tend to leave the code as is and not revisit all type_strings for now but I do not object your suggestion in general. I hope you are ok when we either wait for failures and fix them then or we have another cleanup PR after this one that uses script_run/assert_script_run in more locations rather than type_string

Seems a little bit risky, but depending on the circumstances I might do the same thing.